impl/src/main/java/org/jboss/weld/bean/InterceptorImpl.java

```diff
@@ -125,19 +125,29 @@ public void initializeAfterBeanDiscovery() {
     }
 
     private void checkInterceptorBindings() {
-        if (interceptorMetadata.isEligible(org.jboss.weld.interceptor.spi.model.InterceptionType.POST_CONSTRUCT) ||
-                interceptorMetadata.isEligible(org.jboss.weld.interceptor.spi.model.InterceptionType.PRE_DESTROY) ||
-                interceptorMetadata.isEligible(org.jboss.weld.interceptor.spi.model.InterceptionType.POST_ACTIVATE) ||
-                interceptorMetadata.isEligible(org.jboss.weld.interceptor.spi.model.InterceptionType.PRE_PASSIVATE)) {
+        if (interceptorMetadata.isEligible(org.jboss.weld.interceptor.spi.model.InterceptionType.POST_CONSTRUCT)
+                || interceptorMetadata.isEligible(org.jboss.weld.interceptor.spi.model.InterceptionType.PRE_DESTROY)
+                || interceptorMetadata.isEligible(org.jboss.weld.interceptor.spi.model.InterceptionType.POST_ACTIVATE)
+                || interceptorMetadata.isEligible(org.jboss.weld.interceptor.spi.model.InterceptionType.PRE_PASSIVATE)) {
             for (Annotation interceptorBindingType : interceptorBindingTypes) {
                 Target target = interceptorBindingType.annotationType().getAnnotation(Target.class);
-                if (target != null && Arrays2.unorderedEquals(target.value(), ElementType.TYPE, ElementType.METHOD)) {
-                    throw ReflectionLogger.LOG.methodElementTypeNotAllowed(this, interceptorBindingType.annotationType());
+                if (target == null || hasInvalidTargetType(target.value())) {
```

Comment:
Yes, I agree that we should be consistent. The validation of the special case of `{TYPE, METHOD}` seems to be here since CDI 1.0 times where `CONSTRUCTOR` was not available and all the other cases were ruled out by `InterceptionBindingModel`.

Nevertheless, given that for a long time now Weld allowed almost any combination of ElementTypes to be specified  within the `@Target` annotation (or to not specify `@Target` at all), a breaking change may be dangerous. I think sooner or later we'll find an application/framework which breaks if we go for strict checking.

In addition, the interceptors spec does not require that we throw an exception either. Therefore, I suggest to revisit InterceptionBindingModel checks together with this check and log a warning for each violation (no special treatment for `{TYPE, METHOD}` anymore).